lib/trento/operations/host_policy.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Trento.Operations.HostPolicy do
+  @moduledoc """
+  HostReadModel operation policies
+  """
+
+  @behaviour Trento.Operations.PolicyBehaviour
+
+  alias Trento.Clusters.Projections.ClusterReadModel
+  alias Trento.Databases.Projections.DatabaseInstanceReadModel
+  alias Trento.Hosts.Projections.HostReadModel
+  alias Trento.SapSystems.Projections.ApplicationInstanceReadModel
+
+  # saptune_solution_apply operation authorized when:
+  # - all SAP instances are authorized for maintenance
+  def authorize_operation(
+        :saptune_solution_apply,
+        %HostReadModel{
+          cluster: cluster,
+          application_instances: application_instances,
+          database_instances: database_instances
+        } = host,
+        _
+      ) do
+    applications_maintenance_authorized =
+      application_instances
+      |> Enum.map(fn application_instance ->
+        %ApplicationInstanceReadModel{application_instance | host: host}
```

Comment:
Well, I guess the circular reference requires us to do this 😄

Reply:
Yes to avoid cyclical preloads hehe